src/metrics/kubernetes.py

```diff
@@ -0,0 +1,97 @@
+# Python Imports
+import socket
+import logging
+from typing import List, Tuple
+from kubernetes.client import CoreV1Api, V1PodList
+from kubernetes.stream import portforward
+
+
+logger = logging.getLogger(__name__)
+
+
+class KubernetesManager:
```

Comment:
Given this is sourced from some kubernetes code (as you told me via call), I'd write down somewhere where this was sourced from.

Reply:
Will put in as a link as well, it is the same one.